lib/private/Files/Cache/Wrapper/CacheJail.php

```diff
@@ -222,7 +245,29 @@ private function formatSearchResults($results) {
 	 * @return array an array of file data
 	 */
 	public function search($pattern) {
-		$results = $this->getCache()->search($pattern);
+		// normalize pattern
+		$pattern = $this->normalize($pattern);
+
+		if ($pattern === '%%') {
+			return [];
+		}
+
+		$query = $this->getQueryBuilder();
+		$query->selectFileCache()
+			->whereStorageId()
+			->andWhere($query->expr()->orX(
+				$query->expr()->like('path', $query->createNamedParameter($this->getGetUnjailedRoot() . '/%')),
+				$query->expr()->eq('path_hash', $query->createNamedParameter(md5($this->getGetUnjailedRoot()))),
```

Comment:
```suggestion
				$query->expr()->eq('path_hash', $query->createNamedParameter(md5($this->getGetUnjailedRoot())))
```
To support older php versions.